action/protocol/rewarding/fund.go

```diff
@@ -118,6 +115,10 @@ func (p *Protocol) assertEnoughBalance(
 
 // DepositGas deposits gas into the rewarding fund
 func DepositGas(ctx context.Context, sm protocol.StateManager, amount *big.Int, registry *protocol.Registry) error {
+	// If the gas fee is 0, return immediately
+	if amount.Cmp(big.NewInt(0)) == 0 {
```

Comment:
should we handle negative case?

Reply:
negative amount are asserted in the rewarding protocol